migration.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/python3
+
+import argparse
+import json
+
+import requests
+
+
+def prepare_labels(rule):
+    if "labels" not in rule:
+        return
+
+    labels = rule["labels"]
+    labels.update({
+        "percona_alerting": "1",
+        "severity": rule["severity"].lstrip("SEVERITY_").lower(),
+        "template_name": rule["template_name"]
+    })
+    return labels
+
+
+def prepare_annotations(rule):
+    annotations = rule["annotations"]
+    annotations.update({"rule": rule["name"]})
+
+    return annotations
+
+
+def prepare_expression(rule):
+    expr = rule["expr"]
+
+    if "filters" not in rule:
+        return expr
+
+    for f in rule["filters"]:
+        expr = 'label_match({}, "{}", "{}")'.format(expr, f["key"], f["value"])
+
+    return expr
+
+
+def convert_rule(rule):
+    return {
+        "grafana_alert": {
+            "title": rule["name"] + "_" + rule["rule_id"],
+            "condition": "A",
+            "no_data_state": "OK",
+            "exec_err_state": "Alerting",
+            "data": [
+                {
+                    "refId": "A",
+                    "datasourceUid": datasourceUID,
+                    "relativeTimeRange": {"from": 600, "to": 0},
+                    "model": {
+                        "expr": prepare_expression(rule),
+                        "refId": "A",
+                        "instant": True,
+                    },
+                },
+            ]
+        },
+        "for": rule["for"],
+        "annotations": prepare_annotations(rule),
+        "labels": prepare_labels(rule),
+    }
+
+
+parser = argparse.ArgumentParser(description="Migration script for Integrated Alerting alert rules")
+parser.add_argument("-u", "--user", required=True, help="PMM user login")
+parser.add_argument("-p", "--password", required=True, help="PMM user password")
+parser.add_argument("-s", "--server-url", default="http://localhost/", help="PMM server URL (default: %(default)s)")
+parser.add_argument("-i", "--insecure", action="store_false", help="skip TLS certificates verification")
+parser.add_argument("-f", "--folder", default="Experimental",
+                    help="folder in Grafana where to put migrated alert rules (default: %(default)s)")
+parser.add_argument("-g", "--group", default="migrated",
+                    help="alert group name for migrated alert rules (default: %(default)s)")
+
+config = vars(parser.parse_args())
+auth = (config["user"], config["password"])
+
+# Create alert group for migrated rules
+groupURL = '{server_url}/graph/api/ruler/grafana/api/v1/rules/{folder}/{group}'.format(**config)
+groupReq = requests.get(groupURL, auth=auth, verify=config["insecure"])
```

Comment:
requests.get can produce exceptions that should be handled (e.g. call sys.exit here with some message). The best way is to use try/except here